core/modules/filter/tests/filter_dialog.test

```diff
@@ -0,0 +1,125 @@
+ */
+
+/**
+ * Test for filter dialog link validation.
+ */
+class FilterEditorLinkValidateTestCase extends BackdropWebTestCase {
+
+  protected $profile = 'testing';
+  protected $editorUser;
+  protected $format;
+
+  /**
+   * Set up testing environment.
+   */
+  public function setUp() {
+    parent::setUp(
+      'filter',
+      'ckeditor',
+      'filter_formtest',
+      'language',
+      'locale',
+    );
+
+    // Add a language and set it as default.
+    $german = (object) array(
+      'langcode' => 'de',
+      'name' => 'German',
+      'direction' => LANGUAGE_LTR,
+    );
+    language_save($german);
+    config_set('system.core', 'language_default', 'de');
+
+    // Create text format using filter default settings.
+    $filter_info = filter_filter_info();
+    $filters = array_fill_keys(array_keys($filter_info), array());
+    $format = (object) array(
+      'format' => 'filter_custom',
+      'name' => 'Filter custom',
+      'filters' => $filters,
+      'editor' => 'ckeditor',
+    );
+    filter_format_save($format);
+    $permission = filter_permission_name($format);
+    $this->format = $format;
+
+    // Create content type.
+    $this->backdropCreateContentType(array('type' => 'page', 'name' => 'Page'));
+
+    // Create user and log in.
+    $this->editorUser = $this->backdropCreateUser(array(
+      'access content',
+      'bypass node access',
+      $permission,
+    ));
+    $this->backdropLogin($this->editorUser);
+  }
+
+  /**
+   * Check function _filter_format_editor_link_url_validate().
+   */
+  public function testEditorLinkValidate() {
+    // Create a node and get the nid for the calling path.
+    $node = $this->backdropCreateNode();
+    $calling_path = "node/{$node->nid}/edit";
+    $link_token = filter_editor_dialog_token($this->format, 'link', $this->editorUser, $calling_path);
+    $german = language_load('de');
+    $options = array(
+      'query' => array(
+        'token' => $link_token,
+        'calling_path' => $calling_path,
+      ),
+      'language' => $german,
+    );
+    $this->backdropGet('editor/dialog/link/' . $this->format->format, $options);
+
+    $edit = array(
+      'attributes[text]' => 'The link',
+    );
+    $invalid_urls = array(
+      'javascript://boo',
+      'http://foo>bar.baz',
+      'http://foobar\.baz',
+    );
+    foreach ($invalid_urls as $url) {
+      $edit['attributes[href]'] = $url;
+      $this->backdropPost(NULL, $edit, t('Save'));
+      $this->assertRaw('The URL <em class="placeholder">' . check_plain($url) . '</em> is not valid.', format_string('Invalid URL %url did not pass validation', array(
+        '%url' => $url,
+      )));
+    }
+
+    $valid_urls = array(
+      // Should get altered.
+      'node/1' => '/de/node/1',
+      'ärger-im-büro' => '/de/%C3%A4rger-im-b%C3%BCro',
+      '<front>?page=1#main-content' => '/de?page=1#main-content',
+      // Should stay untouched.
+      '#main-content' => '#main-content',
```

Comment:
Could you add the valid url `/en/node/1` here as one that should stay untouched?

Reply:
I added those.